### ISSUE TITLE
SDL crashes while sending OnSystemRequest in case if RPC service is protected

### ISSUE THREAD

**Issue by dboltovskyi** [smartdevicelink/sdl_core#2922]
### Bug Report
SDL crashes while sending OnSystemRequest in case if RPC service is protected

##### Reproduction Steps
1. Register new app
=> Policy Update is started
2. Switch RPC service to protected mode
=> SDL continues PTU sequence
=> SDL->App: OnSystemRequest(with snapshot)

##### Expected Behavior
SDL proceed with OnSystemRequest notification successfully

##### Observed Behavior
SDL crashes with error:
```
smartDeviceLinkCore: ./sdl_core/src/components/security_manager/src/ssl_context_impl.cc:478: 
virtual bool security_manager::CryptoManagerImpl::SSLContextImpl::Encrypt(const uint8_t*, size_t, const uint8_t**, size_t*): 
Assertion `(len == static_cast<size_t>(read_size))' failed.
```

##### OS & Version Information
* OS/Version: Ubuntu 16.04 amd64
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/release/5.1.1
* ATF: https://github.com/smartdevicelink/sdl_atf/tree/develop

##### Test Case, Sample Code, and / or Example App
[001_Protected_Mode_PTU.lua.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3570087/001_Protected_Mode_PTU.lua.tar.gz)
[2019-05-22_12-28-31_PROPRIETARY.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3209077/2019-05-22_12-28-31_PROPRIETARY.tar.gz)
[2019-05-22_12-28-51_EXTERNAL_PROPRIETARY.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3209078/2019-05-22_12-28-51_EXTERNAL_PROPRIETARY.tar.gz)

**Comment by dboltovskyi** [comment]
@theresalech Please triage this issue. 
It's very likely it could affect https://github.com/smartdevicelink/sdl_core/issues/2816 which has been merged recently.

**Comment by theresalech** [comment]
@dboltovskyi can you please confirm if you're able to reproduce this issue on `develop`?

**Comment by dboltovskyi** [comment]
@theresalech Confirmed. Issue is reproduced on a current develop (e7b5c75). 
Attached script was slightly updated due to changes introduced by `Read Generic Signal Data` feature.

**Comment by theresalech** [comment]
@dboltovskyi okay thank you.  We will investigate.  Do you anticipate submitting a fix for this issue as well?

**Comment by dboltovskyi** [comment]
@theresalech Yes, we're going to submit a fix

**Comment by dboltovskyi** [comment]
@theresalech Please notice PRs for `core` and `scripts` are ready for review

**Comment by iCollin** [comment]
Closed with merge of #3021
